example/fft_ex06_float128.cpp

```diff
@@ -0,0 +1,78 @@
+///////////////////////////////////////////////////////////////////
+//  Copyright Eduardo Quintana 2021
+//  Copyright Janek Kozicki 2021
+//  Copyright Christopher Kormanyos 2021
+//  Distributed under the Boost Software License,
+//  Version 1.0. (See accompanying file LICENSE_1_0.txt
+//  or copy at http://www.boost.org/LICENSE_1_0.txt)
+/*
+    boost::math::fft example 06
+    
+    Fast Polynomial Multiplication with quadmath
+*/
+
+#include <boost/math/fft/bsl_backend.hpp>
+#include <boost/multiprecision/float128.hpp>
+#include <iostream>
+#include <vector>
+#include <exception>
+
+template<class T>
+std::vector<T> multiply_complex(const std::vector<T>& A, const std::vector<T>& B)
+{
+  const std::size_t N = A.size();
+  std::vector< boost::multiprecision::complex<T> > TA(N),TB(N);
```

Comment:
I am still worried about this line. We are using boost multiprecision headers.
@cosurgi @ckormanyos :If you agree I can take the opportunity to remove those dependencies as well in this PR, consequently having to completely remove the complex selector,
and this template (the complex) has to be filled by hand for every real type on the user side.

Reply:
Hi @Lagrang3 

At this stage, I would recommend to merge this PR.

> worried about this line. We are using boost multiprecision headers

I would recommend to investigate another potentially viable solution. Actually Boost.Math _does_ have its own version of a generic-complex. But we have solely used it for quadmath complex support up to this time point, if memory serves me correctly.

Boost.Math's own complex can be found within the framework of `<boost/cstdfloat.hpp>`, such as [here](https://github.com/boostorg/math/blob/develop/include/boost/cstdfloat.hpp). Now going further, at [this line](https://github.com/boostorg/math/blob/d8fe127b827e496c0dc8546d419cdda3aa2e0a00/include/boost/cstdfloat.hpp#L49), we find the inclusion of another header which eventually leads us [here](https://github.com/boostorg/math/blob/develop/include/boost/math/cstdfloat/cstdfloat_complex_std.hpp).

There we find an implementation of a generalized `std::complex` --- potentially strongly controversial due to the use of namespace `std`.

I would ask you to find if using this particular complex instead of Multiprecision's adaption(s) works out. I'm not immediately sure if you'd need to `#define` or even un-`#define` something to get it to compile. The advantage is that this complex resides within Boost.Math entirely. The disadvantage is placing more demand and upkeep on the `cstdfloat` part of Boost.Math, an area which I am already having difficulty properly supporting due to time constraints.

Cc: @cosurgi

Reply:
@Lagrang3 

All of the above discussion regarding `<boost/cstdfloat.hpp>` ultimately grants access to a Boost.Math type `boost::math::cstdfloat::detail::float_internal128_t` which has support for limits, `<cmath>`-like functions and can be used in a homegrown version of `std::complex`. It does not deal with Multiprecision's complex adaptions, potentially leading to further confusion. But you'll get a quad complex in Math out of the deal.

@cosurgi

Reply:
Let me investigate.

Reply:
For example 06 we don't actually need to deduce a complex. We just want to show how to use the real fft interface.
On the other hand, I am tempted to re-use [cstdfloat_complex_std](https://github.com/boostorg/math/blob/develop/include/boost/math/cstdfloat/cstdfloat_complex_std.hpp) to substitute the [simple_complex](https://github.com/BoostGSoC21/math/blob/develop/include/boost/math/fft/simple_complex.hpp). But I guess it will collide with `std::complex`.

Reply:
And yes, It would be nice to use it, since it is already written and inside Boost.Math.